src/Task.ts

```diff
@@ -82,4 +83,8 @@ export function empty<A>(): Task<A> {
   return never as Task<A>
 }
 
+export function fromPromise<L, A>(f: Lazy<Promise<A>>, onrejected: (reason: any) => L): Task<Either<L, A>> {
+  return new Task(() => f().then(a => right<L, A>(a), reason => left<L, A>(onrejected(reason))))
+}
+
```

Comment:
Perhaps `tryCatch`, as the `async` part is implicit with the `Task` module?

So then we'll have `either.tryCatch => Either` and `task.tryCatch => Task<Either>>`?